File: python/palindrome.py
def check_palindrome(string):
	if len(string) > 1:
		if string[0] == string[-1]:
			check_palindrome(string[1:][:-1])
		else:
			print "Not a palindrome"
	else:
		print "Palindrome"

text = raw_input("Please enter your text string")
check_palindrome(text.lower().replace(" ",""))

File: scala/palindrome.scala
def palindrome(string : String) : Unit = {
	if(string.length > 1)
	{
		if(string(0) == string(string.length - 1))
			{
				palindrome(string.substring(1, string.length  - 1))
			}
		else
			{
				println("Not a palindrome")
			}
	}
	else
	{
		println("Palindrome")	
	}


}

println("Please enter your required text")
var string = readLine().toLowerCase.replace(" ", "")
//println(string)
palindrome(string)
